catalog/async/catalog-async-data/src/main/java/org/codice/ddf/catalog/async/data/impl/LazyProcessResourceImpl.java

```diff
@@ -159,8 +179,13 @@ public boolean isModified() {
   }
 
   @Override
-  public void close() {
-    IOUtils.closeQuietly(inputStream);
+  public synchronized void close() {
+    try {
+      if (streamCloser != null) {
+        streamCloser.close();
+      }
+    } catch (IOException e) {
+    } // suppress exceptions
   }
```

Comment:
❗️ I would suggest making `populateProcessResource()` and `loadResource()` also synchronized and make the following variables that are updated by the methods and read by other methods not synchronized volatile: `size`, `name`, and `mimeType`.